Feat(eos_designs): Add eBGP LAN HA support for CV Pathfinder (#3494)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_designs/docs/wan-preview.md

  - Policies are assigned to VRFs using the list `wan_virtual_topologies.vrfs`. A policy can be reused in multiple VRFs.
  - If no policy is assigned for the `default` VRF policy, AVD auto generates one with one `default_virtual_topology` entry configured to use all available local path-groups.
  - For the policy defined for VRF `default` (or the auto-generared one), an extra match statement is injected in the policy to match the traffic towards the Pathfinders or AutoVPN RRs, the name of the application-profile is hardcoded as `CONTROL-PLANE-APPLICATION-PROFILE`. A special policy is created by appending `-WITH-CP` at the end of the targetted policy name.

>   - For HA, the considered interfaces are only the `uplink_interfaces` in VRF default. It is possible to disable HA at the site level.

Reviewer: ```suggestion
  - For HA, the considered interfaces are only the `uplink_interfaces` in VRF default. It is possible to disable HA under node settings.
```

###### HA

for eBGP LAN routing protocol the following is done to enable HA:

> - the HA interface(s) is(are) the uplink interface(s).

Reviewer: ```suggestion
- the uplink interfaces are used as HA interfaces.
```

> - the HA interface(s) subnet(s) are redistributed in BGP via the `RM-CONN-2-BGP` route-map

Reviewer: ```suggestion
- the subnets of the HA interfaces are to BGP via the `RM-CONN-2-BGP` route-map
```

Author: ```suggestion
- the subnets of the HA interfaces are redistributed to BGP via the `RM-CONN-2-BGP` route-map
```

> - BGP underlay peer group is configured with `allowas-in 1` to be able to learn the HA peer uplink interface subnet coming with the same ASN via eBGP over the LAN.

Reviewer: ```suggestion
- BGP underlay peer group is configured with `allowas-in 1` to be able to learn the HA peer uplink interface subnet over the LAN as well as learning WAN routes from other sites (as backup in case all wan links are lost)
```

Author: ```suggestion
BGP underlay peer group is configured with `allowas-in 1` to be able to learn the HA peer uplink interface subnet over the LAN as well as learning WAN routes from other sites (as backup in case all WAN links are lost).
```

- the Underlay peer group is configured with two route-maps

>   - one outbound route-map `RM-WAN-HA-SOO-OUT` to tag all the WAN received routes with the SOO `<bgp_as>:<wan_site_id>` except the uplink interface subnet.

Reviewer: ```suggestion
  - one inbound route-map `RM-UNDERLAY-PEERS-IN`
    - Match HA peer's uplink subnets (not marked) to be able to form HA tunnel (not exported to EVPN).
    - Match HA peer's originated prefixes, set longer AS path and mark with SoO to export to EVPN. These will be used as backup from other sites to destinations on HA Peer Router in case all WAN connections on Peer are down.
    - Match all WAN routes using AS path and set no-advertise community. This will be used as backup routes to the WAN in case this router looses all WAN connections.
    - Match anything else (LAN prefixes) and mark with the SoO `<bgp_as>:<wan_site_id>` to export to EVPN.
```

>   - one inbound route-map `RM-WAN-HA-SOO-IN` denying any route with the SOO.

Reviewer: ```suggestion
  - one outbound route-map `RM-UNDERLAY-PEERS-OUT`
    - allowing local routes marked with SoO (routes/interfaces defined via tenants + router-id)
    - allowing subnets of uplink interfaces.
    - allow all routes learned from iBGP (WAN)
    - Implicitly denying other routes which could be learned from BGP towards a WAN provider or redistributed without marking with SoO.
```
